test/distributed/optim/test_named_optimizer.py

```diff
@@ -0,0 +1,245 @@
+            lr=1e-2,
+            momentum=0.6,
+        )
+
+        for _ in range(2):
+            x = torch.rand(5, 8)
+            y = m(x)
+            y.sum().backward()
+            named_optim_2.step()
+
+        state_dict_before_load = named_optim_2.state_dict()
+
+        # Compare "state" in optim state dict
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net1.0.weight"],
+            state_dict_before_load["state"]["net1.0.weight"],
+            assert_equal=False,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net2.0.bias"],
+            state_dict_before_load["state"]["net2.0.bias"],
+            assert_equal=False,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net3.weight"],
+            state_dict_before_load["state"]["net3.weight"],
+            assert_equal=False,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net4.1.bias"],
+            state_dict_before_load["state"]["net4.1.bias"],
+            assert_equal=False,
+        )
+
+        named_optim_2.load_state_dict(state_dict_to_load)
+        state_dict_after_load = named_optim_2.state_dict()
+
+        # Compare "state" in optim state dict
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net1.0.weight"],
+            state_dict_after_load["state"]["net1.0.weight"],
+            assert_equal=True,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net2.0.bias"],
+            state_dict_after_load["state"]["net2.0.bias"],
+            assert_equal=True,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net3.weight"],
+            state_dict_after_load["state"]["net3.weight"],
+            assert_equal=True,
+        )
+        self._compare_state_dict_group(
+            state_dict_to_load["state"]["net4.1.bias"],
+            state_dict_after_load["state"]["net4.1.bias"],
+            assert_equal=True,
+        )
+
+    def test_load_state_dict_error(self):
+        m = TestDummyModel()
+        named_optim_1 = _NamedOptimizer(
+            m.named_parameters(),
+            torch.optim.SGD,
+            lr=1e-2,
+            momentum=0.9,
+        )
+
+        for _ in range(2):
+            x = torch.rand(5, 8)
+            y = m(x)
+            y.sum().backward()
+            named_optim_1.step()
+
+        state_dict_to_load = named_optim_1.state_dict()
+
+        named_optim_2 = _NamedOptimizer(
+            m.named_parameters(),
+            torch.optim.SGD,
+            lr=1e-2,
+            momentum=0.6,
+        )
+
+        err_msg = (
+            "Expects the optim to be initialized before load but found not initialized"
+        )
+        with self.assertRaisesRegex(ValueError, err_msg):
+            named_optim_2.load_state_dict(state_dict_to_load)
```

Comment:
nit: Add newline (might be required by linter).

Reply:
Linter does not complain previously. But adding it now.